src/gl_limits.rs

```diff
@@ -29,26 +43,87 @@ impl Serialize for GLLimits {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
         where S: Serializer
     {
-        [self.max_vertex_attribs, self.max_tex_size, self.max_cube_map_tex_size]
+        [self.max_vertex_attribs, self.max_tex_size, self.max_cube_map_tex_size,
+         self.max_combined_texture_image_units, self.max_fragment_uniform_vectors,
+         self.max_renderbuffer_size, self.max_texture_image_units, self.max_varying_vectors,
+         self.max_vertex_texture_image_units, self.max_vertex_uniform_vectors]
             .serialize(serializer)
     }
 }
 
+macro_rules! gl_integer {
+    ($gl:ident, $pname:ident) => {
+        {
+            let mut val = [0];
+            unsafe {
+                $gl.get_integer_v(gl::$pname, &mut val);
+            }
+            assert_eq!($gl.get_error(), gl::NO_ERROR, "Error retrieving {}", stringify!($pname));
```

Comment:
Maybe just `debug_assert_eq!`, though not big deal :)

Reply:
I think it's important to catch this in release builds too.